src/intersection-observer.js

```diff
@@ -81,20 +81,29 @@ export function getIntersectionChangeEntry(element, owner, viewport) {
       'Negative dimensions in element.');
   // Building an IntersectionObserverEntry.
 
-  let intersectionRect = element;
+  // The element is relative to (0, 0), while the viewport moves. So, we must
+  // adjust.
+  const boundingClientRect = moveLayoutRect(element, -viewport.left,
```

Comment:
Yes, `FixedLayer` can be used here. `Viewport` currently has `isDeclaredFixed` method that responds with "this element could be fixed". But I think we can dig a bit deeper and expose "this element is currently fixed" result.

Overall, is this fix critical? Would it wait for layers in any form?